org-node/src/client.rs

```diff
@@ -235,11 +285,200 @@ impl Client {
+                                Err(err) => {
+                                    tracing::error!(target: "org-node", "Error setting head: {}", err);
+                                    ack.send(Err(err)).ok();
+                                    continue;
+                                }
+                                Ok(()) => {}
+                            }
+
+                            if let Err(err) = Client::sign_refs(urn, &api).await {
+                                tracing::error!(target: "org-node", "Error signing refs: {}", err);
+                                ack.send(Err(err)).ok();
+                                continue;
+                            }
+
+                            // Acknowledge success;
+                            ack.send(Ok(response)).ok();
+                        }
+                        Ok(None) => {
+                            tracing::error!(target: "org-node", "Project head not found! Cannot set head for peer");
+                            ack.send(Err(Error::ProjectHeadNotFound)).ok();
+                        }
+                        Err(err) => {
+                            tracing::error!(target: "org-node", "Error getting project head: {}", err);
+                            ack.send(Err(err)).ok();
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    async fn track_peer_handler(
+        peer_tx: mpsc::Sender<ProjectPeerTransmitter>,
+        project_peer: ProjectPeer,
+    ) -> Result<Option<PeerId>, Error> {
+        // create receiver for Project Peer acknowledgement.
+        let (tx, rx) = oneshot::channel::<Result<Option<PeerId>, Error>>();
+
+        // spawn rx listener to wait for oneshot response from peer_tx.
+        let rx_res = tokio::spawn(rx);
+
+        // notify track peers process to handle tracking and fetching.
+        if let Err(e) = peer_tx
+            .send((
+                ProjectPeer {
+                    api: project_peer.api,
+                    urn: project_peer.urn,
+                    peer_info: project_peer.peer_info,
+                    paths: project_peer.paths,
+                },
+                tx,
+            ))
+            .await
+        {
+            tracing::error!(target: "org-node", "Failed to send project peer request: {}", e);
+            return Err(Error::ProjectPeer(e.to_string()));
+        }
+
+        // Await response from project peer `track_peer` process;
+        match rx_res.await {
+            Ok(ack) => match ack {
+                Ok(Err(e)) => {
+                    tracing::error!(target: "org-node", "Project peer returned error: {}", e);
+                    Err(e)
+                }
+                Err(e) => {
+                    tracing::error!(target: "org-node", "Error project peer receiver failed: {}", e);
+                    Err(Error::Recv(e))
+                }
+                Ok(res) => res,
+            },
```

Comment:
Any way to avoid these nested result matches? It's a bit hard to follow.

Reply:
5dfbfd6 removes nested match arms